### PR TITLE
Refactor Makefile and Dockerfile

### PR DESCRIPTION
**Pull request by erbesharat**
**What this PR does / why we need it**:
Refactor Makefile and Dockerfile

Remaining:

- Add the project's path dynamically to the Dockerfile (Line 4 and 6)
- Add push target to the makefile
- ...

**Special notes for your reviewer**:
I will add whatever comes to my mind to the list above, let me know if we need to add something else or change something.

### CODE REVIEW

**Comment by stp-ip** [comment]
So a few things already:
* currently this means we are shipping the golang build container within the txtdirect container, which seems overhead, that is unnecessary
* building a alpine based image, with just the runtime needed assets (for now the binary, later the binary + tor) makes this easier
* why use a build container with a hardcoded docker file? Couldn't we pass in the go build command to a build container and mount in the working directory and then pull in deps within the container and build it?
* same idea could be used for testing, but this might make testing locally a bit harder as docker build be needed.

As long as in both cases we use the make file as source of truth it doesn't really matter.

Travis:
"docker run -v $(pwd):/src golang:1.11-alpine /bin/sh -c "make test" <- testing stage (uses make dep under the hood)
"docker run -v $(pwd):/src golang:1.11-alpine /bin/sh -c "make deploy" <- deploy stage (uses make dep and make build under the hood)

Dev:
make test and/or make build

Happy to iterate with your thoughts.

**Comment by stp-ip** [comment]
The other option is to require docker being installed to run the build. This gives us the ability to build with the latest image we provide and don't have to care much about the users install. As we are not going with using the full build template, it seems the middle ground with having make as the source of truth and using docker for isolation within travis is ok.

**Comment by stp-ip** [comment]
Adding to this. Using docker for both the ci and the user makes the overall interaction with changed code less likely. A test will always run with fresh deps so switching between projects or changing something locally never is an issue as local deps are never relied upon. This requires more resources/bandwidth overall so I understand that this might not be perfect, which brings us back to the middle ground, where we have the option to built and test with local deps, but it's also easy before the final push to do a clean run inside docker and the same can be used within travis.

So maybe have the option to use docker for any command, which just mounts the sourcecode and runs the make command within a container.
make docker test -> docker run .... make test

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,7 @@
-FROM golang:1.11-alpine
+FROM alpine:3.9
 
-RUN apk --no-cache add ca-certificates make git && \
-  mkdir -p $GOPATH/src/github.com/txtdirect/txtdirect
+RUN apk --no-cache add ca-certificates
 
-WORKDIR $GOPATH/src/github.com/txtdirect/txtdirect
+ADD txtdirect /caddy
```
Comment (stp-ip) [inline Dockerfile]: Why add it as `/caddy`?

Comment (erbesharat) [reply Dockerfile]: It was `/caddy` in the previous version, so I thought maybe there is a reason for it :smile:

Comment (stp-ip) [reply Dockerfile]: Not that I know of. Might just have been an oversight from before :)

**Pull request merged**